cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,249 @@
+            + "|.*-909[1-4]\\.sasl\\.enabled\\.mechanisms"
+            + "|advertised\\.listeners"
+            + "|zookeeper\\.connect"
+            + "|broker\\.rack)$");
+
+    public KafkaBrokerConfigurationDiff(Config brokerConfigs, String desired, KafkaVersion kafkaVersion, int brokerId) {
+        this.configModel = KafkaConfiguration.readConfigModel(kafkaVersion);
+        this.brokerId = brokerId;
+        this.diff = computeDiff(brokerId, desired, brokerConfigs, configModel);
+    }
+
+    private static void fillPlaceholderValue(Map<String, String> orderedProperties, String placeholder, String value) {
+        orderedProperties.entrySet().forEach(entry -> {
+            entry.setValue(entry.getValue().replaceAll("\\$\\{" + Pattern.quote(placeholder) + "\\}", value));
+        });
+    }
+
+    /**
+     * Returns true if property in desired map has a default value
+     * @param key name of the property
+     * @param config Config which contains property
+     * @return true if property in desired map has a default value
+     */
+    boolean isDesiredPropertyDefaultValue(String key, Config config) {
+        Optional<ConfigEntry> entry = config.entries().stream().filter(configEntry -> configEntry.name().equals(key)).findFirst();
+        if (entry.isPresent()) {
+            return entry.get().isDefault();
+        }
+        return false;
+    }
+
+    public boolean canBeUpdatedDynamically() {
+        boolean result = true;
+        for (AlterConfigOp entry : diff) {
+            if (isEntryReadOnly(entry.configEntry())) {
+                result = false;
+            }
+        }
+        return result;
+    }
+
+    /**
+     * @param entry tested ConfigEntry
+     * @return true if the entry is READ_ONLY
+     */
+    private boolean isEntryReadOnly(ConfigEntry entry) {
+        return configModel.get(entry.name()).getScope().equals(Scope.READ_ONLY);
+    }
+
+    /**
+     * @return A map which can be used for dynamic configuration of kafka broker
+     */
+    public Map<ConfigResource, Collection<AlterConfigOp>> getConfigDiff() {
+        return Collections.singletonMap(Util.getBrokersConfig(brokerId), diff);
+    }
+
+    /**
+     * @return The number of broker configs which are different.
+     */
+    public int getDiffSize() {
+        return diff.size();
+    }
+
+    private static boolean isIgnorableProperty(String key) {
+        return IGNORABLE_PROPERTIES.matcher(key).matches();
+    }
+
+    /**
+     * Computes diff between two maps. Entries in IGNORABLE_PROPERTIES are skipped
+     * @param brokerId id of compared broker
+     * @param desired desired configuration
+     * @param brokerConfigs current configuration
+     * @param configModel default configuration for {@code kafkaVersion} of broker
+     * @return KafkaConfiguration containing all entries which were changed from current in desired configuration
+     */
+    private static Collection<AlterConfigOp> computeDiff(int brokerId, String desired,
```

Comment:
```suggestion
    private static Collection<AlterConfigOp> diff(int brokerId, String desired,
```
I think that it is inferred that a call to diff would compute a diff, just a. recommended change